app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -357,71 +396,126 @@ public void onViewCreated(@NonNull final View view, @Nullable final Bundle saved
+        mMapView.getZoomController().setVisibility(CustomZoomButtonsController.Visibility.NEVER);
+        mMapView.getController().setZoom(ZOOM_LEVEL);
+        MapTileProviderBasic tileProvider = new MapTileProviderBasic(this.getContext());
+        ITileSource tileSource = new XYTileSource(
+            "FietsRegionaal",
+            3,
+            18,
+            256,
+            ".png",
+            new String[]{"http://overlay.openstreetmap.nl/openfietskaart-rcn/"}
+        );
+        tileProvider.setTileSource(tileSource);
+        tileProvider.getTileRequestCompleteHandlers().add(mMapView.getTileRequestCompleteHandler());
+        TilesOverlay tilesOverlay = new TilesOverlay(tileProvider, this.getContext());
+        tilesOverlay.setLoadingBackgroundColor(Color.TRANSPARENT);
+        mMapView.getOverlays().add(tilesOverlay);
+        mMapView.getOverlays().add(new MapEventsOverlay(new MapEventsReceiver() {
+            @Override
+            public boolean singleTapConfirmedHelper(GeoPoint p) {
+                if (isListBottomSheetExpanded()) {
+                    // Back should first hide the bottom sheet if it is expanded
+                    hideBottomSheet();
+                } else if (isDetailsBottomSheetVisible()) {
+                    hideBottomDetailsSheet();
                 }
-                performMapReadyActions();
-                final CameraPosition cameraPosition;
-                if(applicationKvStore.getString("LastLocation")!=null) { // Checking for last searched location
-                    String[] locationLatLng = applicationKvStore.getString("LastLocation").split(",");
-                    cameraPosition = new CameraPosition.Builder()
-                        .target(new LatLng(Double.valueOf(locationLatLng[0]),
-                            Double.valueOf(locationLatLng[1])))
-                        .zoom(ZOOM_LEVEL)
-                        .build();
-                }else {
-                    cameraPosition = new CameraPosition.Builder()
-                        .target(new LatLng(51.50550, -0.07520))
-                        .zoom(ZOOM_OUT)
-                        .build();
+                return false;
+            }
+
+            @Override
+            public boolean longPressHelper(GeoPoint p) {
+                return false;
+            }
+        }));
+
+        mMapView.addMapListener(new MapListener() {
+            @Override
+            public boolean onScroll(ScrollEvent event) {
+                if (lastMapFocus != null) {
+                    Location mylocation = new Location("");
+                    Location dest_location = new Location("");
+                    dest_location.setLatitude(mMapView.getMapCenter().getLatitude());
+                    dest_location.setLongitude(mMapView.getMapCenter().getLongitude());
+                    mylocation.setLatitude(lastMapFocus.getLatitude());
+                    mylocation.setLongitude(lastMapFocus.getLongitude());
```

Comment:
I killed+restarted the app many times, but it still shows me (the blue dot) at the place where I was 4 hours ago.
I wonder if something here may need to be updated?
The app has always been a bit slow to update the location, but killing+restarting the app always fixed it.

Reply:
I have fixed this issue and tested it using android emulator in newer commit
https://drive.google.com/file/d/1ylx3JjyFRUlAS1pRmzn-Kypgz7eFKFvb/view?usp=sharing